addons/pos_self_order/static/src/ProductList/ProductList.js

```diff
@@ -0,0 +1,217 @@
+/** @odoo-module */
+
+import { Component, onMounted, useEffect, useRef, useState } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/SelfOrderService";
+import { useAutofocus } from "@web/core/utils/hooks";
+import { formatMonetary } from "@web/views/fields/formatters";
+import { NavBar } from "@pos_self_order/NavBar/NavBar";
+import { fuzzyLookup } from "@web/core/utils/search";
+export class ProductList extends Component {
+    static template = "ProductList";
+    static components = { NavBar };
+    setup() {
+        this.state = useState(this.env.state);
+        this.privateState = useState({
+            selectedTag: "",
+            searchIsFocused: false,
+            searchInput: "",
+            navbarIsShown: true,
+            scrollingDown: false,
+            scrolling: false,
+        });
+        this.selfOrder = useSelfOrder();
+        this.formatMonetary = formatMonetary;
```

Comment:
create getters for formatted values you want to show in the template. No need to provide `formatMonetary` in the component.

Reply:
E.g. `formatMonetary(product.list_price, {currencyId: selfOrder.config.currency_id})`.

```js
getFormattedListPrice(product) {
  return formatMonetary(product.list_price, {currencyId: selfOrder.config.currency_id})
}
```

use `getFormattedListPrice` in the template.

But this is just my personal preference. No need to actually follow.